src/components/DateInput.jsx

```diff
@@ -117,7 +117,7 @@ export default class DateInput extends React.Component {
           value={value}
           onChange={this.onChange}
           onKeyDown={this.onKeyDown}
-          onFocus={onFocus}
+          //onFocus={onFocus}
```

Comment:
Hmm, so it was a keyboard navigation (probably right-left arrows?), if yes, the problem occurred during swiping in Mac (using Ctrl+leftArrow/rightArrow). As an advice, you can use a controller, which will remove listener on yours keyUp event and add a default keyUp (if arrows are user with Ctrl), and reverse in case if arrows was used separately... something like that.